Failed to compiled with GCC 14

```
code/renderer_vulkan/tr_bsp.c:1228:29: warning: passing argument 1 to ‘restrict’-qualified parameter aliases with argument 2 [-Wrestrict]
 1228 |                 memcpy( grid->heightLodError, grid->heightLodError, grid->height * 4 );
      |                         ~~~~^~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~
code/renderer_vulkan/tr_bsp.c: In function ‘RE_LoadWorldMap’:
code/renderer_vulkan/tr_bsp.c:1840:27: error: passing argument 2 of ‘ri.FS_ReadFile’ from incompatible pointer type [-Wincompatible-pointer-types]
 1840 |     ri.FS_ReadFile( name, &buffer );
      |                           ^~~~~~~
      |                           |
      |                           char **
code/renderer_vulkan/tr_bsp.c:1840:27: note: expected ‘void **’ but argument is of type ‘char **’
make[2]: *** [Makefile:2887: build/release-linux-ppc64le/renderer_vulkan/tr_bsp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
code/renderer_vulkan/tr_fonts.c: In function ‘RE_RegisterFont’:
code/renderer_vulkan/tr_fonts.c:385:38: error: passing argument 2 of ‘ri.FS_ReadFile’ from incompatible pointer type [-Wincompatible-pointer-types]
  385 |                 ri.FS_ReadFile(name, &faceData);
      |                                      ^~~~~~~~~
      |                                      |
      |                                      char **
code/renderer_vulkan/tr_fonts.c:385:38: note: expected ‘void **’ but argument is of type ‘char **’
make[2]: *** [Makefile:2887: build/release-linux-ppc64le/renderer_vulkan/tr_fonts.o] Error 1
code/renderer_vulkan/tr_image.c: In function ‘RE_RegisterSkin’:
code/renderer_vulkan/tr_image.c:200:27: error: passing argument 2 of ‘ri.FS_ReadFile’ from incompatible pointer type [-Wincompatible-pointer-types]
  200 |     ri.FS_ReadFile( name, &text );
      |                           ^~~~~
      |                           |
      |                           char **
code/renderer_vulkan/tr_image.c:200:27: note: expected ‘void **’ but argument is of type ‘char **’
make[2]: *** [Makefile:2887: build/release-linux-ppc64le/renderer_vulkan/tr_image.o] Error 1
code/renderer_vulkan/R_FindShader.c: In function ‘ScanAndLoadShaderFiles’:
code/renderer_vulkan/R_FindShader.c:544:58: error: passing argument 2 of ‘ri.FS_ReadFile’ from incompatible pointer type [-Wincompatible-pointer-types]
  544 |                 long summand = ri.FS_ReadFile( filename, &buffers[i] );
      |                                                          ^~~~~~~~~~~
      |                                                          |
      |                                                          char **
```

The new GCC 14 is a bit stricter. Trivial bug to fix.

--- a/code/renderer_vulkan/R_FindShader.c
+++ b/code/renderer_vulkan/R_FindShader.c
@@ -541,7 +541,7 @@ void ScanAndLoadShaderFiles( void )
 
 		snprintf( filename, sizeof( filename ), "scripts/%s", shaderFiles[i] );
 		ri.Printf( PRINT_ALL, "...loading '%s'\n", filename );
-		long summand = ri.FS_ReadFile( filename, &buffers[i] );
+		long summand = ri.FS_ReadFile( filename, (void**)&buffers[i] );
 		
 		if ( !buffers[i] )
 			ri.Error( ERR_DROP, "Couldn't load %s", filename );

--- a/code/renderer_vulkan/R_ImageBMP.c
+++ b/code/renderer_vulkan/R_ImageBMP.c
@@ -69,7 +69,7 @@ void R_LoadBMP( const char *name, byte **pic, int *width, int *height )
 	//
 	// load the file
 	//
-	length = ri.FS_ReadFile( name, &buffer);
+	length = ri.FS_ReadFile( name, (void**)&buffer );
 	if (!buffer || length < 0) {
 		return;
 	}

--- a/code/renderer_vulkan/R_ImageJPG.c
+++ b/code/renderer_vulkan/R_ImageJPG.c
@@ -113,7 +113,7 @@ void R_LoadJPG(const char *filename, unsigned char **pic, int *width, int *heigh
    * requires it in order to read binary files.
    */
 
-  len = ri.FS_ReadFile ( filename, &fbuffer);
+  len = ri.FS_ReadFile ( filename, (void**)&fbuffer);
   if (!fbuffer || len < 0) {
 	return;
   }

--- a/code/renderer_vulkan/R_ImagePCX.c
+++ b/code/renderer_vulkan/R_ImagePCX.c
@@ -69,7 +69,7 @@ void R_LoadPCX ( const char *filename, byte **pic, int *width, int *height)
 	//
 	// load the file
 	//
-	int len = ri.FS_ReadFile( filename, &raw);
+	int len = ri.FS_ReadFile( filename, (void**)&raw );
 	if (!raw || len < 0) {
 		return;
 	}

--- a/code/renderer_vulkan/R_ImagePNG.c
+++ b/code/renderer_vulkan/R_ImagePNG.c
@@ -250,7 +250,7 @@ static struct BufferedFile *ReadBufferedFile(const char *name)
 	 *  Read the file.
 	 */
 
-	BF->Length = ri.FS_ReadFile(name, &buffer);
+	BF->Length = ri.FS_ReadFile(name, (void**)&buffer);
 	BF->Buffer = (unsigned char*)buffer;
 
 	/*

--- a/code/renderer_vulkan/R_ImageTGA.c
+++ b/code/renderer_vulkan/R_ImageTGA.c
@@ -61,7 +61,7 @@ void R_LoadTGA ( const char *name, unsigned char** pic, int *width, int *height)
 	//
 	// load the file
 	//
-	length = ri.FS_ReadFile ( name, &buffer);
+	length = ri.FS_ReadFile ( name, (void**)&buffer);
 	if (!buffer || length < 0) {
 		return;
 	}

--- a/code/renderer_vulkan/R_LoadImage2.c
+++ b/code/renderer_vulkan/R_LoadImage2.c
@@ -156,7 +156,7 @@ static void LoadTGA( const char* name, unsigned char** pic, uint32_t* width, uin
 	//
 	// load the file
 	//
-	ri.FS_ReadFile(name, &buffer);
+	ri.FS_ReadFile(name, (void**)&buffer);
 	if (!buffer) {
 		return;
 	}
@@ -384,7 +384,7 @@ static void LoadTGA( const char* name, unsigned char** pic, uint32_t* width, uin
 static void LoadJPG( const char* name, unsigned char** pic, uint32_t* width, uint32_t* height)
 {
     char* fbuffer;
-    int len = ri.FS_ReadFile(name, &fbuffer);
+    int len = ri.FS_ReadFile(name, (void**)&fbuffer);
     if (!fbuffer) {
         return;
     }
@@ -459,7 +459,7 @@ static void LoadBMP( const char *name, unsigned char **pic, uint32_t *width, uin
 	//
 	// load the file
 	//
-	length = ri.FS_ReadFile(name, &buffer);
+	length = ri.FS_ReadFile(name, (void**)&buffer);
 	if (!buffer) {
 		return;
 	}
@@ -625,7 +625,7 @@ static void LoadPCX ( const char *filename, unsigned char **pic, unsigned char *
 	//
 	// load the file
 	//
-	len = ri.FS_ReadFile(filename, &raw);
+	len = ri.FS_ReadFile(filename, (void**)&raw);
 	if (!raw) {
 		return;
 	}

--- a/code/renderer_vulkan/R_LoadMD3.c
+++ b/code/renderer_vulkan/R_LoadMD3.c
@@ -169,7 +169,7 @@ qhandle_t R_RegisterMD3(const char* name, model_t* mod)
 	char* buf;
 	int	numLoaded = 0;
 
-	ri.FS_ReadFile( name, &buf );
+	ri.FS_ReadFile( name, (void**)&buf );
 	
     if( NULL != buf)
     {
@@ -219,7 +219,7 @@ qhandle_t R_RegisterMD3(const char* name, model_t* mod)
 		char namebuf[MAX_QPATH+20] = {0};
         snprintf(namebuf, sizeof(namebuf), "%s_%d.md3", filename, lod);
 
-		ri.FS_ReadFile( namebuf, &buf );
+		ri.FS_ReadFile( namebuf, (void**)&buf );
 		if(!buf)
 			continue;
 		

--- a/code/renderer_vulkan/R_LoadMDR.c
+++ b/code/renderer_vulkan/R_LoadMDR.c
@@ -496,7 +496,7 @@ qhandle_t R_RegisterMDR(const char *name, model_t *mod)
 	char* buf;
 
 	qboolean loaded = qfalse;
-	int filesize = ri.FS_ReadFile(name, &buf);
+	int filesize = ri.FS_ReadFile(name, (void**)&buf);
 	if(!buf)
 	{
 		mod->type = MOD_BAD;

--- a/code/renderer_vulkan/tr_bsp.c
+++ b/code/renderer_vulkan/tr_bsp.c
@@ -1837,7 +1837,7 @@ void RE_LoadWorldMap( const char *name )
 
 
 	// load it
-    ri.FS_ReadFile( name, &buffer );
+	ri.FS_ReadFile( name, (void**)&buffer );
 	if ( !buffer ) {
 		ri.Error (ERR_DROP, "RE_LoadWorldMap: %s not found", name);
 	}

--- a/code/renderer_vulkan/tr_fonts.c
+++ b/code/renderer_vulkan/tr_fonts.c
@@ -382,7 +382,7 @@ void RE_RegisterFont(const char *fontName, int pointSize, fontInfo_t *font)
 
 	int len = ri.FS_ReadFile(name, NULL);
 	if (len == sizeof(fontInfo_t)) {
-		ri.FS_ReadFile(name, &faceData);
+		ri.FS_ReadFile(name, (void**)&faceData);
 		fdOffset = 0;
 		fdFile = (unsigned char*)faceData;
 		for(i=0; i<GLYPHS_PER_FONT; i++) {

--- a/code/renderer_vulkan/tr_image.c
+++ b/code/renderer_vulkan/tr_image.c
@@ -197,7 +197,7 @@ qhandle_t RE_RegisterSkin( const char *name )
 	}
 
 	// load and parse the skin file
-    ri.FS_ReadFile( name, &text );
+    ri.FS_ReadFile( name, (void**)&text );
 	if ( !text ) {
 		return 0;
 	}

--- a/code/renderer_vulkan/tr_model_iqm.c
+++ b/code/renderer_vulkan/tr_model_iqm.c
@@ -1146,7 +1146,7 @@ qhandle_t R_RegisterIQM(const char *name, model_t *mod)
 	qboolean loaded = qfalse;
 	int filesize;
 
-	filesize = ri.FS_ReadFile(name, &buf);
+	filesize = ri.FS_ReadFile(name, (void**)&buf);
 	if(!buf)
 	{
 		mod->type = MOD_BAD;

